### PR TITLE
Implement search functionality and improve UI

### PR DESCRIPTION
**Pull request by zachseidner1**
# Changes made
- Add search bar component (test it in the stats page)
  - Includes filter card and results card
  - View all search results is a TODO since we need a design for this
- Fix happiness number issues, update component to be of type number
- Add auto selection to entries page so it will automatically given a start date

### CODE REVIEW

File: src/pages/Statistics/SearchBar.tsx [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+  // Add keyboard listener for pressing enter and navigation between entries
+  useEffect(() => {
+    // handle when user presses enter to make a search or change selected entry
+    // Initialization of this function must be done in use effect to prevent
+    // data from being capturerd in closure
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === "Enter") {
+        if (resultsShowing && selectedEntryIndex >= 0 && data && selectedEntryIndex < data.length) {
+          window.open(BASE_FRONTEND_URL + `/home?date=${data[selectedEntryIndex].timestamp}`);
```
Comment (jonathanjma) [inline src/pages/Statistics/SearchBar.tsx:82]: prolly should use relative url, take a look at this https://github.com/jonathanjma/happiness-app-frontend/blob/1391942c6db273c43c7b8e6cc639a61b80c8a270/src/pages/UserGroups/GroupCard.tsx#L39

File: src/pages/Statistics/SearchResult.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      <div className="flex flex-grow min-w-[32px]" />
+      {!selected && <label className="text-gray-600 mr-4 min-w-[100px]">{parseYYYYmmddFormat(happiness.timestamp).toLocaleDateString("en-us", {
+        month: "short",
+        day: "numeric",
+        year: "numeric"
+      })}</label>}
+      {selected &&
+        <button className="bg-gray-50 rounded-lg p-1 h-10 mr-4 min-w-[120px]"
+          onClick={() => {
+            window.open(BASE_FRONTEND_URL + `/home?date=${happiness.timestamp}`);
```
Comment (jonathanjma) [inline src/pages/Statistics/SearchResult.tsx:38]: see previous comment

**Review by jonathanjma — CHANGES_REQUESTED** [review]

File: src/pages/Statistics/SearchBar.tsx (hunk 1 above)
Comment (zachseidner1) [reply src/pages/Statistics/SearchBar.tsx:82]: I didn't know this was a thing, thanks!

**Review by jonathanjma — APPROVED** [review]
:-)

**Pull request merged**